bindings/nodejs/lib/gpio.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ var soletta = require( 'bindings' )( 'soletta' ),
+    _ = require( 'lodash' );
+
+var edges = {
+    "none": soletta.sol_gpio_edge.SOL_GPIO_EDGE_NONE,
+    "rising": soletta.sol_gpio_edge.SOL_GPIO_EDGE_RISING,
+    "falling": soletta.sol_gpio_edge.SOL_GPIO_EDGE_FALLING,
+    "any": soletta.sol_gpio_edge.SOL_GPIO_EDGE_BOTH,
+}
+
+function pin_read_cb( pin, value ) {
+    _pins[ pin ].dispatchEvent( "change", {
+                        type: "change",
+                        value: value
+    } );
+}
```

Comment:
Why not make this function enclosed in the context it is used, then you'd have enough context to dispatch the event, not requiring to use the array.